middleware/publishers/klipper.py

```
@@ -327,29 +308,20 @@ def _handle_toolhead(self, moonraker: str, event: SpoolEvent) -> bool:
-                    timeout=5,
-                ).raise_for_status()
+                send_gcode(
+                    moonraker,
+                    f"SAVE_VARIABLE VARIABLE={event.target.lower()}_spool_id VALUE={event.spool_id}",
+                )
             except Exception:
                 # Rollback: revert Spoolman to prevent orphaned spool_id (#15)
                 logger.error(
                     "[toolhead] SAVE_VARIABLE failed for spool %s on %s — rolling back Spoolman",
                     event.spool_id, event.target,
                 )
```

Ruff (0.15.20)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)